app/src/main/java/com/eveningoutpost/dexdrip/NavDrawerBuilder.java

```diff
@@ -69,26 +69,31 @@ public NavDrawerBuilder(final Context context) {
             if (is_active_sensor) {
                 if (!CollectionServiceStarter.isBTShare(context)) {
                     if (last_two_bgReadings.size() > 1 || Ob1G5CollectionService.isG5WantingCalibration()) {
-                        if ((last_two_calibrations.size() > 1) && !Ob1G5CollectionService.isG5WantingInitialCalibration()) {
+                        if ((last_two_calibrations.size() > 1) && !Ob1G5CollectionService.isG5WantingInitialCalibration()) { //This only happens after initial calibration.
                             // TODO tighten this time limit
                             if (bGreadings_in_last_30_mins.size() >= 2) {
                                 long time_now = JoH.tsl();
                                 if ((time_now - last_two_calibrations.get(0).timestamp < (1000 * 60 * 60))
-                                && !Ob1G5CollectionService.isG5WantingCalibration()) { //Put steps in place to discourage over calibration
+                                        && !Ob1G5CollectionService.isG5WantingCalibration()) { //Put steps in place to discourage over calibration
                                     this.nav_drawer_options.add(context.getString(R.string.override_calibration));
                                     this.nav_drawer_intents.add(new Intent(context, CalibrationOverride.class));
-                                } else {
+                                } else { //G5, old G6, or Firefly, in no-code mode, after initial calibration and long enough after previous calibration
                                     this.nav_drawer_options.add(context.getString(R.string.add_calibration));
                                     this.nav_drawer_intents.add(new Intent(context, AddCalibration.class));
                                 }
-                            } else {
+                            } else {  //G5, old G6 or Firefly, in no-code mode, not long after a calibration
                                 this.nav_drawer_options.add(context.getString(R.string.cannot_calibrate_right_now));
                                 this.nav_drawer_intents.add(new Intent(context, Home.class));
                             }
-                        } else {
+                        } else { //If the has been no two initial calibrations
                             if (BgReading.isDataSuitableForDoubleCalibration() || Ob1G5CollectionService.isG5WantingInitialCalibration()) {
-                                this.nav_drawer_options.add(context.getString(R.string.initial_calibration));
-                                this.nav_drawer_intents.add(new Intent(context, DoubleCalibrationActivity.class));
+                                if (last_two_bgReadings.size() > 1) { //Firefly G6, in factory calibration mode, after third reading
```

Comment:
I have made changes since this question was asked.  And the question was being asked from someone else.
Is there still a question, for me, after the changes?